themes/bootstrap3/templates/RecordDriver/DefaultRecord/list-entry.phtml

```diff
@@ -85,7 +85,11 @@
                   echo '<strong>' . $this->transEsc($snippet['caption']) . ':</strong> ';
                 }
                 if (!empty($snippet['snippet'])) {
-                  echo '<span class="quotestart">&#8220;</span>...' . $this->highlight($snippet['snippet']) . '...<span class="quoteend">&#8221;</span><br/>';
+                  echo '<span class="quotestart">&#8220;</span>'
```

Comment:
Using *eol*_ellipsis at the middle of the line feels not quite right. Maybe we should have a `highlight_snippet_html` translation that contains all this at least up to the `<br>`? For instance:

`highlight_snippet_html = '<span class="quotestart">&#8220;</span>…%%snippet%%…<span class="quoteend">&#8221;</span>';`

?

Reply:
Note: I'd included the spans in the translation string because I think the quotes should be considered part of it, but if you disagree, the most important part would be `…%%snippet%%…`.

Reply:
Can't check this right now, but does it allow for language differences? In German, we use ellipsis without space before the ellipsis when we truncate a word but there must be a space before the ellipsis if we truncate a sentence or phrase, like so

ellips... 
BUT
ellipses are cool and ...

Reply:
Well, at least it wouldn't make the situation worse if we take the default form from eol_ellipsis, though might need some adjustment so that the preceding ellipsis has the right spacing as well, if that's desired. This is assuming all backends cut the snippet by word boundary, otherwise there's no way of doing it perfectly all the time.

Reply:
Thanks, @EreMaijala, I agree that this is a good idea, and I'll implement it as soon as time permits. Regarding @ckaz's question, @EreMaijala's approach is clearly better than the concatenation version I chose, because as currently implemented here, it will be wrong in German -- `" ...snippet ..."` instead of `"... snippet ..."`. Creating a snippet translation gives us the control we need to do it correctly. This will currently have to be implemented to assume that the snippet ends on word boundaries on both ends, which I suppose may not always be true if different backends implement things differently... but based on the current behavior of the system, we don't have enough information to be more nuanced. Hopefully it's "good enough for now" at least. :-)

Reply:
Okay, please take a look at [9947180](https://github.com/vufind-org/vufind/pull/2381/commits/9947180855286edf489936726fc065b1435d5375) -- this adds highlight_snippet_html as proposed by @EreMaijala, and it seems to work reasonably well for me. A couple of thoughts, though:

1.) I'm not especially comfortable with including CSS classes in the language files; I prefer to keep things as simple as possible. Given that the quotestart and quoteend classes do not appear to be used in any of our CSS, and they were presumably put there to make it possible to customize the appearance of snippets, how would you feel about removing the spans entirely and perhaps replacing the HTML entities with their Unicode equivalents? I think that might feel like a better fit for a language file, and I don't think it would reduce the ability to customize the appearance. I would still keep the _html suffix on the language string, since we need to be sure not to escape it in order to properly render the highlighting in the injected snippet.

2.) I'm a bit puzzled as to why we have snippet logic in templates related to display favorite lists. I don't see how there could ever be active snippets in this context. I'm wondering if this was a long-ago copy-paste error, or if there's some actual intention behind it....

Reply:
1.) That'd make sense. I think that making it possible to style the whole thing easier would be more important for customization than targeting the quotes alone. This could be simply enclosing the whole snippet thing in `<div class="highlight-snippet">...</div>` (which would probably eliminate the need for a trailing `<br>`, but would need to check for preceding stuff). That's something way out of scope for this PR. At least now you can add them to the translation even though that doesn't help with the caption.

2.) I'd bet it was just copied without further thought. That's not the only odd thing, though. It doesn't make sense to me to check for a snippet caption before checking for a snippet, particularly as the caption ends with a colon.

Reply:
Thanks, @EreMaijala!

1.) I've simplified the language strings with [7182f63](https://github.com/vufind-org/vufind/pull/2381/commits/7182f633283eea8db0144c90a39394f7c5d82f9c). Note that I used "straight" instead of "curly" quotes for the RTL languages to ensure that things don't display strangely (I noticed some inconsistent results when displaying LTR snippets within the RTL context... so using more generic quotes seemed like the easiest solution to avoid a big project). I agree that adding a wrapping `<div>` for styling could be beneficial, but I'm considering that out of scope for this PR.

2.) I'll leave the list-entry.phtml stuff alone for now since it's not hurting anything and changing it seems out of scope for this chunk of work. After merging this, I can open a follow-up PR to remove it, if you think that's worthwhile.

Reply:
Thanks, looks good to me now!

2.) I don't think it's worth it alone, but could be done if we decide to do other tweaks as well.